apis/v1alpha2/shared_types.go

```diff
@@ -186,14 +186,57 @@ type BackendRef struct {
 // RouteConditionType is a type of condition for a route.
 type RouteConditionType string
 
+// RouteConditionReason is a reason for a route condition.
+type RouteConditionReason string
+
 const (
 	// This condition indicates whether the route has been accepted or rejected
 	// by a Gateway, and why.
-	ConditionRouteAccepted RouteConditionType = "Accepted"
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Accepted"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	RouteConditionAccepted RouteConditionType = "Accepted"
+
+	// FIXME: alias for backwards compatibility in v1alpha2, remove in next release
+	ConditionRouteAccepted RouteConditionType = RouteConditionAccepted
```

Comment:
We've generally had a policy that breaking changes/renames to our Go types/APIs are acceptable in any release as long as they don't result in breaking changes to the CRDs. That matches the upstream behavior, so we could just remove these lines. I'm also open to changing this to be stricter.